src/include/execution/tplclass.h

```diff
@@ -0,0 +1,422 @@
+
+#include "loggers/loggers_util.h"
+#include "settings/settings_manager.h"
+
+#define __SETTING_GFLAGS_DEFINE__      // NOLINT
+#include "settings/settings_common.h"  // NOLINT
+#include "settings/settings_defs.h"    // NOLINT
+#undef __SETTING_GFLAGS_DEFINE__       // NOLINT
+
+// ---------------------------------------------------------
+// CLI options
+// ---------------------------------------------------------
+
+llvm::cl::OptionCategory kTplOptionsCategory("TPL Compiler Options",
+                                             "Options for controlling the TPL compilation process.");
+llvm::cl::opt<std::string> kInputFile(llvm::cl::Positional, llvm::cl::desc("<input file>"), llvm::cl::init(""),
+                                      llvm::cl::cat(kTplOptionsCategory));
+llvm::cl::opt<bool> kPrintAst("print-ast", llvm::cl::desc("Print the programs AST"),
+                              llvm::cl::cat(kTplOptionsCategory));
+llvm::cl::opt<bool> kPrintTbc("print-tbc", llvm::cl::desc("Print the generated TPL Bytecode"),
+                              llvm::cl::cat(kTplOptionsCategory));
+llvm::cl::opt<std::string> kOutputName("output-name", llvm::cl::desc("Print the output name"),
+                                       llvm::cl::init("schema1"), llvm::cl::cat(kTplOptionsCategory));
+llvm::cl::opt<bool> kIsSQL("sql", llvm::cl::desc("Is the input a SQL query?"), llvm::cl::cat(kTplOptionsCategory));
+
+tbb::task_scheduler_init scheduler;
+
+namespace terrier::execution {
+
+    class TplClass {
+    public:
+        static constexpr const char *kExitKeyword = ".exit";
+
+        // Terrier objects
+        terrier::transaction::TransactionManager * txn_manager_pointer_;
+        exec::SampleOutput * sample_output_pointer_;
+        terrier::catalog::db_oid_t db_oid_;
+        terrier::catalog::Catalog * catalog_pointer_;
+        bool *unfinished_;
+
+        static const int max_num_modules_ = 6;
+        std::map <std::string, int> name_id_;
+        std::unique_ptr <vm::Module> modules_[max_num_modules_];
+        int num_modules_;
+
+        TplClass(terrier::transaction::TransactionManager * txn_manager_pointer,
+                 exec::SampleOutput * sample_output_pointer,
+                 terrier::catalog::db_oid_t db_oid,
+                 terrier::catalog::Catalog * catalog_pointer,
+                 bool *unfinished) :
+                txn_manager_pointer_(txn_manager_pointer),
+                sample_output_pointer_(sample_output_pointer),
+                db_oid_(db_oid),
+                catalog_pointer_(catalog_pointer),
+                unfinished_(unfinished) {
+            num_modules_ = 0;
+            name_id_.clear();
+        }
+
+/**
+ * Compile the TPL source in \a source and run it in both interpreted and JIT
+ * compiled mode
+ * @param source The TPL source
+ * @param name The name of the module/program
+ */
+        //void CompileAndRun(const std::string &source, const std::string &name = "tmp-tpl") {
+
```

Comment:
Create a separate cpp file to put all the functions below there:
https://github.com/cmu-db/terrier/wiki/Coding-Guideline#h-and-cpp-files

And then use forward declaration to remove as much header includes as possible:
https://github.com/cmu-db/terrier/wiki/Coding-Guideline#forward-declaration